os/net/link-stats.c

```diff
@@ -98,30 +99,30 @@ link_stats_is_fresh(const struct link_stats *stats)
 }
 /*---------------------------------------------------------------------------*/
 #if LINK_STATS_INIT_ETX_FROM_RSSI
+/*
+ * Returns initial ETX value from an RSSI value.
+ *    RSSI >= RSSI_HIGH           -> use default ETX
+ *    RSSI_LOW < RSSI < RSSI_HIGH -> ETX is a linear function of RSSI
+ *    RSSI <= RSSI_LOW            -> use minimal initial ETX
+ **/
 uint16_t
 guess_etx_from_rssi(const struct link_stats *stats)
 {
   if(stats != NULL) {
     if(stats->rssi == LINK_STATS_RSSI_UNKNOWN) {
       return ETX_DEFAULT * ETX_DIVISOR;
     } else {
-      /* A rough estimate of PRR from RSSI, as a linear function where:
-       *      RSSI >= -60 results in PRR of 1
-       *      RSSI <= -90 results in PRR of 0
-       * prr = (bounded_rssi - RSSI_LOW) / (RSSI_DIFF)
-       * etx = ETX_DIVOSOR / ((bounded_rssi - RSSI_LOW) / RSSI_DIFF)
-       * etx = (RSSI_DIFF * ETX_DIVOSOR) / (bounded_rssi - RSSI_LOW)
-       * */
-#define ETX_INIT_MAX 3
-#define RSSI_HIGH -60
-#define RSSI_LOW  -90
-#define RSSI_DIFF (RSSI_HIGH - RSSI_LOW)
-      uint16_t etx;
-      int16_t bounded_rssi = stats->rssi;
-      bounded_rssi = MIN(bounded_rssi, RSSI_HIGH);
-      bounded_rssi = MAX(bounded_rssi, RSSI_LOW + 1);
-      etx = RSSI_DIFF * ETX_DIVISOR / (bounded_rssi - RSSI_LOW);
-      return MIN(etx, ETX_INIT_MAX * ETX_DIVISOR);
+#define RSSI_DIFF (LINK_STATS_RSSI_HIGH - LINK_STATS_RSSI_LOW)
+      CTASSERT(RSSI_DIFF > 0 && "RSSI_HIGH must be greater then RSSI_LOW");
+      CTASSERT(ETX_DIVISOR * RSSI_DIFF < 0x10000 && "RSSI math overflow");
```

Comment:
This is subjective but, in my humble opinion, the defines within the function greatly reduces the readability of the code. I would much prefer it being at the top of the file or in the header-file.

Reply:
yeah, I don't disagree with this comment. will update the PR